types/coin.go

```diff
@@ -26,44 +33,44 @@ func (coin Coin) SameDenomAs(other Coin) bool {
 
 // IsZero returns if this represents no money
 func (coin Coin) IsZero() bool {
-	return coin.Amount == 0
+	return coin.Amount.IsZero()
 }
 
 // IsGTE returns true if they are the same type and the receiver is
 // an equal or greater value
 func (coin Coin) IsGTE(other Coin) bool {
-	return coin.SameDenomAs(other) && (coin.Amount >= other.Amount)
+	return coin.SameDenomAs(other) && (!coin.Amount.LT(other.Amount))
 }
 
 // IsEqual returns true if the two sets of Coins have the same value
 func (coin Coin) IsEqual(other Coin) bool {
-	return coin.SameDenomAs(other) && (coin.Amount == other.Amount)
+	return coin.SameDenomAs(other) && (coin.Amount.Equal(other.Amount))
 }
 
 // IsPositive returns true if coin amount is positive
 func (coin Coin) IsPositive() bool {
-	return (coin.Amount > 0)
+	return (coin.Amount.Sign() == 1)
 }
 
 // IsNotNegative returns true if coin amount is not negative
 func (coin Coin) IsNotNegative() bool {
-	return (coin.Amount >= 0)
+	return (coin.Amount.Sign() != -1)
```

Comment:
Can we add a helper function to `sdk.Int` for this, e.g. `sdk.Int.IsNotNegative`?